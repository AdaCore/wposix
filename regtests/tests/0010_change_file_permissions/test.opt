windows-XP DEAD no icacls on this platform
windows-2012 XFAIL
