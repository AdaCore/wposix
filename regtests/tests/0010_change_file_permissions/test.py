
from test_support import *
import os

USER=os.getenv("USER")

def remove_acl(file):
    exec_cmd('icacls', [file, '/remove', 'Everyone'], output_file='acl.log')
    exec_cmd('icacls', [file, '/remove', 'None'], output_file='acl.log')
    exec_cmd('icacls', [file, '/remove', 'cyg_server'], output_file='acl.log')
    exec_cmd('icacls', [file, '/remove', 'SYSTEM'], output_file='acl.log')
    exec_cmd('icacls', [file, '/remove', 'Administrators'],
             output_file='acl.log')
    exec_cmd('icacls', [file, '/remove', USER], output_file='acl.log')

def create(file):
    f=open(file, 'w')
    f.close()
    remove_acl(file)

gnatmake('cperms')

create('check1')

exec_cmd('icacls', ['check1', '/grant', 'Everyone:R'], output_file='acl.log')
exec_cmd('icacls', ['check1', '/grant', USER+':R'], output_file='acl.log')
exec_cmd('icacls', ['check1', '/grant', 'None:R'], output_file='acl.log')

run('cperms')
