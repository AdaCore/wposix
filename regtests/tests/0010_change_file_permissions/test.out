File: check1
OTHERS_EXECUTE = FALSE
OTHERS_WRITE = FALSE
OTHERS_READ = TRUE
GROUP_EXECUTE = FALSE
GROUP_WRITE = FALSE
GROUP_READ = TRUE
OWNER_EXECUTE = FALSE
OWNER_WRITE = FALSE
OWNER_READ = TRUE
SET_GROUP_ID = FALSE
SET_USER_ID = FALSE

File: check1
OTHERS_EXECUTE = FALSE
OTHERS_WRITE = FALSE
OTHERS_READ = TRUE
GROUP_EXECUTE = FALSE
GROUP_WRITE = TRUE
GROUP_READ = TRUE
OWNER_EXECUTE = FALSE
OWNER_WRITE = TRUE
OWNER_READ = TRUE
SET_GROUP_ID = FALSE
SET_USER_ID = FALSE

File: check1
OTHERS_EXECUTE = FALSE
OTHERS_WRITE = FALSE
OTHERS_READ = FALSE
GROUP_EXECUTE = FALSE
GROUP_WRITE = FALSE
GROUP_READ = FALSE
OWNER_EXECUTE = TRUE
OWNER_WRITE = TRUE
OWNER_READ = TRUE
SET_GROUP_ID = FALSE
SET_USER_ID = FALSE

File: check1
OTHERS_EXECUTE = FALSE
OTHERS_WRITE = FALSE
OTHERS_READ = FALSE
GROUP_EXECUTE = TRUE
GROUP_WRITE = TRUE
GROUP_READ = TRUE
OWNER_EXECUTE = FALSE
OWNER_WRITE = FALSE
OWNER_READ = FALSE
SET_GROUP_ID = FALSE
SET_USER_ID = FALSE

File: check1
OTHERS_EXECUTE = TRUE
OTHERS_WRITE = TRUE
OTHERS_READ = TRUE
GROUP_EXECUTE = FALSE
GROUP_WRITE = FALSE
GROUP_READ = FALSE
OWNER_EXECUTE = FALSE
OWNER_WRITE = FALSE
OWNER_READ = FALSE
SET_GROUP_ID = FALSE
SET_USER_ID = FALSE

File: check1
OTHERS_EXECUTE = TRUE
OTHERS_WRITE = TRUE
OTHERS_READ = TRUE
GROUP_EXECUTE = TRUE
GROUP_WRITE = TRUE
GROUP_READ = TRUE
OWNER_EXECUTE = TRUE
OWNER_WRITE = TRUE
OWNER_READ = TRUE
SET_GROUP_ID = FALSE
SET_USER_ID = FALSE
